File: soal7.py
def plusMinus(arr):
    n = len(arr)
    positive = len([x for x in arr if x > 0]) / n
    negative = len([x for x in arr if x < 0]) / n
    zero = len([x for x in arr if x == 0]) / n

    # Print hasil dengan 6 angka setelah koma desimal
    print(f"{positive:.6f}")
    print(f"{negative:.6f}")
    print(f"{zero:.6f}")

# Contoh penggunaan
if __name__ == "__main__":
    n = int(input("Masukkan ukuran array: ").strip())
    arr = list(map(int, input().split()))
    plusMinus(arr)

File: soal7.ts
function plusMinus(arr: number[]): void {
    const n: number = arr.length;
    
    const positive: number = arr.filter(x => x > 0).length / n;
    const negative: number = arr.filter(x => x < 0).length / n;
    const zero: number = arr.filter(x => x === 0).length / n;

    // Print hasil dengan 6 angka setelah koma
    console.log(positive.toFixed(6));
    console.log(negative.toFixed(6));
    console.log(zero.toFixed(6));
}

// Contoh penggunaan
const n: number = parseInt(prompt("Masukkan ukuran array: ") || "0");
const arr: number[] = prompt("Masukkan elemen array (dipisah spasi): ")?.split(" ").map(Number) || [];

plusMinus(arr);
